### PR TITLE
V22F-484 Separate Icons

### PR DESCRIPTION
**Pull request by Meaganium**

### CODE REVIEW

File: speed-dreams/source-2.2.3/src/modules/graphic/ssggraph/grboard.cpp [hunk 1]
```diff
@@ -17,30 +17,30 @@
  *                                                                         *
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/modules/graphic/ssggraph/grboard.cpp:17]: Please undo all formatting done to this file except the lines you changed

File: speed-dreams/source-2.2.3/src/modules/graphic/ssggraph/grboard.h [hunk 2]
```diff
@@ -45,29 +45,31 @@ class cGrBoard
...
 
     // SIMULATED DRIVING ASSISTANCE: add display indicator methods
-    void DispIndicators();
-    void DispIndicatorIcon(tTextureData* p_data, ssgSimpleState* p_texture);
-    void DispIndicatorText(tTextData* p_data);
+    void DispActiveIndicators(tIndicator settings, InterventionType interventionType);
+    void DispNeutralIndicators(tIndicator settings, InterventionType interventionType);
+    void DispIndicatorsHelper(tIndicatorData m_indicator, tIndicator settings);
+    void DispIndicatorIcon(tTextureData *p_data, ssgSimpleState *p_texture);
+    void DispIndicatorText(tTextData *p_data);
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/modules/graphic/ssggraph/grboard.h:61]: This is one of our added functions, and thus should follow our coding standard, so undo the change moving the *

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/ConfigEnums.h [hunk 3]
```diff
@@ -29,13 +29,16 @@ typedef unsigned int InterventionType;
 /// @brief The different actions that can be performed
 typedef unsigned int InterventionAction;
 
-#define INTERVENTION_ACTION_NONE       0
+#define INTERVENTION_ACTION_STEER_NONE 0
 #define INTERVENTION_ACTION_TURN_LEFT  1
 #define INTERVENTION_ACTION_TURN_RIGHT 2
 #define INTERVENTION_ACTION_BRAKE      3
 #define INTERVENTION_ACTION_ACCELERATE 4
+#define INTERVENTION_ACTION_BRAKE_NONE 5
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/ConfigEnums.h:37]: You have a STEER_NONE and BRAKE_NONE, why not an ACCELERATE_NONE ?

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/frontend/IndicatorConfig.cpp [hunk 4]
```diff
@@ -49,12 +52,63 @@ std::vector<tIndicatorData> IndicatorConfig::GetActiveIndicators(InterventionTyp
...
+    return first;
+}
+
 /// @brief          Activates the given intervention indicator
 /// @param p_action The intervention to activate the indicators for
 void IndicatorConfig::ActivateIndicator(InterventionAction p_action)
 {
     // TODO: add to the vector instead of overwriting it, this also requires
     //       a way to remove the indicator after some time has passed.
+    std::vector<tIndicatorData> m_indicatorData = IndicatorConfig::GetInstance()->GetIndicatorData();
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/frontend/IndicatorConfig.cpp:111]: Not sure what this is trying to achieve, GetIndicatorData returns m_indicatorData which you have access to directly from here. Also a local variable should not have the m_ prefix.

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]
Couple of issues, for example when accelerating it says brake (the sound) and when trying to steer and accelerate it displays as if it is not steering (try the client blackbox)

Also undo all formatting done to speed dreams file unless they are lines you touched

File: speed-dreams/source-2.2.3/data/data/indicators/config.xml [hunk 5]
```diff
@@ -1,83 +1,168 @@
...
-            <section name="text">
-                <attstr name ="content"   val="Accel"/>
-                <attnum name ="xpos"      val="0.75"/>
-                <attnum name ="ypos"      val="0"/>
-                <attstr name ="font"      val="jura-normal.glf"/>
-                <attnum name ="font-size" val="300"/>
-            </section>
-         </section>
-    </section>
+	<section name="Interventions">
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/data/data/indicators/config.xml:4]: Remove tabs from file. Also in my opinion the disticntion between the intervention type should be made at the top level of the xml (like section=shared_control, etc) instead of having texture1, texture2, etc. This also enables us to add other sounds or text for each intervention type easily, without having to name them 'sound1, sound2, etc', which is not clear at all.

**Review by larsdekwant — CHANGES_REQUESTED** [review]
- Why is there an empty file added called `t.cpp` ? I suppose that is a mistake?
- If we are going to seperate the icons visually, we should also separate them in the code, like having indicators for steer, accel, brake. Each of these can then have sub actions like steering will have none, left and right. Brake can have its own 'none' and 'active', etc.

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/ConfigEnums.h (hunk 3 above)
Comment (Meaganium) [reply speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/ConfigEnums.h:38]: Omdat accelerate en brake allebei onder speed control vallen, dus het is niet logisch om die apart te hebben.

**Review by larsdekwant — APPROVED** [review]
Took over responsibility for this PR from Meagan.
- Implemented all previous requested changes
- Refactored code
- TODO: replace textures by final version (unsure whether that will happen in this PR?)

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/frontend/IndicatorConfig.h [hunk 6]
```diff
@@ -63,13 +68,15 @@ class IndicatorConfig
     IndicatorConfig() = default;
     static IndicatorConfig* m_instance;
 
+    // A vector containing indicator data of every single intervetion action, including the neutral actions.
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/frontend/IndicatorConfig.h:71]:
```suggestion
    // A vector containing indicator data of every single intervention action, including the neutral actions.
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/frontend/IndicatorConfig.h [hunk 7]
```diff
@@ -63,13 +68,15 @@ class IndicatorConfig
...
     std::vector<tIndicatorData> m_indicatorData = std::vector<tIndicatorData>(NUM_INTERVENTION_ACTION);
 
+    // A vector containing the active indiators per intervention action type, indexed by an IndicatorActionType.
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/frontend/IndicatorConfig.h:74]:
```suggestion
    // A vector containing the active indicators per intervention action type, indexed by an IndicatorActionType.
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/frontend/IndicatorConfig.cpp [hunk 8]
```diff
@@ -109,13 +113,13 @@ tScreenPosition IndicatorConfig::LoadScreenPos(void* p_handle, const char* p_pat
 /// @param p_handle The p_handle to the config.xml file
 /// @param p_path   The p_path to the current intervention action to load
 /// @return         The pointer to struct containing the texture data
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/frontend/IndicatorConfig.cpp:115]: Missing @param

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/frontendUnitTests/IndicatorConfigTests.cpp [hunk 9]
```diff
@@ -145,19 +145,18 @@ class IndicatorConfigLoadingTests : public ::testing::Test
...
         }
-
         return data;
     }
 
     /// @brief        Writes the given indicator data object to a xml file.
     /// @param p_data The indicator data
     /// @return       The filepath of the xml file that it has been written to.
-    const char* WriteIndicatorDataToXml(std::vector<tIndicatorData> p_data)
+    const char* WriteIndicatorDataToXml(std::vector<tIndicatorData> p_data, InterventionType p_interventionType)
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/frontendUnitTests/IndicatorConfigTests.cpp:159]: Missing @param

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]

**Review by suppergerrie2 — APPROVED** [review]
